index.js

```diff
@@ -127,9 +127,34 @@ function toggleDetails(details: Element, open: boolean) {
   }
 }
 
+function loadIncludeFragment(event: Event) {
+  const details = event.currentTarget
+  if (!(details instanceof Element)) return
+  const dialog = details.querySelector('details-dialog')
+  if (!(dialog instanceof DetailsDialogElement)) return
+  const state = initialized.get(dialog)
+  if (!state) return
+
+  if (!dialog.src) return
+  if (state.loading || state.loaded) return
+
+  const loader: any = dialog.querySelector('include-fragment')
+  if (loader) {
+    state.loading = true
+    loader.addEventListener('loadend', () => {
+      state.loaded = true
+      state.loading = false
+      autofocus(dialog)
+    })
+    loader.src = dialog.src
+  }
+}
+
 type State = {|
   details: ?Element,
-  activeElement: ?Element
+  activeElement: ?Element,
+  loading?: boolean,
+  loaded?: boolean
```

Comment:
Tracking loaded and loading state might not be needed. An include-fragment is in the loading state when it has a src attribute and is in the loaded state after replacing itself with the server content.